### PR TITLE
add a switch for metricscache

### PR DESCRIPTION
**Pull request by huijunw**
Added a switch to turn on/off metricscache. Config options:
```
heron.topology.metricscachemgr.mode: cluster
heron.topology.metricscachemgr.mode: disabled (default)
```

In some scenarios, the customers do not turn on healthmgr/dhalion, so the metricscache does not have consumer. This PR add a switch so that user can turn it off.

### CODE REVIEW

**Comment by Yaliang** [comment]
Just a general question: is there a test to test configuration settings?

File: heron/executor/src/python/heron_executor.py [hunk 1]
```diff
@@ -238,6 +238,8 @@ def init_parsed_args(self, args):
     self.checkpoint_manager_classpath = parsed_args.checkpoint_manager_classpath
     self.checkpoint_manager_port = parsed_args.checkpoint_manager_port
     self.stateful_config_file = parsed_args.stateful_config_file
+    self.metricscache_manager_mode = parsed_args.metricscache_manager_mode \
+        if parsed_args.jvm_remote_debugger_ports else "disabled"
```
Comment (nwangtw) [inline heron/executor/src/python/heron_executor.py]: why "if parsed_args.jvm_remote_debugger_ports"? typo?

File: heron/spi/src/java/com/twitter/heron/spi/common/Context.java [hunk 2]
```diff
@@ -339,4 +339,9 @@ public static final String statefulStorageCustomClassPath(Config cfg) {
     Object o = statefulStorageConfig.get(Key.STATEFUL_STORAGE_CUSTOM_CLASSPATH.value());
     return o == null ? "" : (String) o;
   }
+
+  public static String metricscacheMgrMode(Config cfg) {
+    return cfg.getStringValue(Key.METRICSCACHEMGR_MODE);
```
Comment (nwangtw) [inline heron/spi/src/java/com/twitter/heron/spi/common/Context.java]: do we need a default value?

Comment (huijunw) [reply heron/spi/src/java/com/twitter/heron/spi/common/Context.java]: there is a default value in Key.java

File: heron/executor/src/python/heron_executor.py (hunk 1 above)
Comment (huijunw) [reply heron/executor/src/python/heron_executor.py]: typo, updated

**Pull request merged**